src/main/java/io/fusionauth/domain/provider/LoginMethod.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2021, FusionAuth, All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+package io.fusionauth.domain.provider;
+
+// TODO : Code Review :
+//    1. Thinking this should be in the provider package instead of oauth2.
+//    2. Maybe rename enum to IdentityProviderLoginMethod, this is more verbose but we are beginning to realize
+//       that because other client libraries we build such as Go and .NET core use either a global namespace or functionally have a
+//       global namespace that naming objects more specifically helps us not have downstream name collisions in our clients.
+// [brettp]TODO: Change name, but may need to rework a ton of tests.
+
+/**
+ * @author Brett Pontarelli
+ */
+public enum LoginMethod {
+  UsePopup,
+  UseRedirect
+}
```

Comment:
I tried this, but lots of things exploded 💥.  I might cut another branch and try again.  I think one thing I noticed is you end up with things like `identityProvider.identityProviderLoginMethod` in some places.

Reply:
This doesn't work ? `public LoginMethod loginMethod` when declared in a class?

Reply:
I wanted to know whether both the `enum` and the var name needed to be changed.  But, as discussed, just the `enum` needs to be changed to `IdentityProviderLoginMethod` and the var can stay `loginMethod`.  I'll update that shortly.